tools/check_format.py

```diff
@@ -147,37 +147,61 @@ def checkNamespace(file_path):
   return []
 
 
-def fixJavaProtoOptions(file_path):
-  package_name = None
-  for line in fileinput.FileInput(file_path):
-    if line.startswith("package "):
-      result = PROTO_PACKAGE_REGEX.search(line)
-      if result is None or len(result.groups()) != 1:
-        continue
+# If the substring is not found in the file, then insert to_add
+def insertProtoOptionIfNotFound(substring, file_path, to_add):
 
-      package_name = result.group(1)
-    if "option java_package = \"io.envoyproxy.envoy" in line:
+  def repl(m):
+    return m.group(0).rstrip() + "\n\n" + to_add + "\n"
+
+  with open(file_path, "r+") as f:
+    text = f.read()
+    if substring in text:
       return []
 
-  if package_name is None:
-    return ["Unable to find package name for proto file: %s" % file_path]
+    new_text = re.sub(PROTO_PACKAGE_REGEX, repl, text)
+    f.seek(0)
```

Comment:
I think you could do this as two separate `with` blocks, one with mode `r` and the other with `w`. This would be conceptually cleaner.